lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -31,9 +31,10 @@ class HiddenContentTransformer(BlockStructureTransformer):
     IMPORTANT: Must be run _after_ the DateOverrideTransformer from edx-when
     in case the 'due' date on a block has been shifted for a user.
     """
-    WRITE_VERSION = 3
-    READ_VERSION = 3
+    WRITE_VERSION = 4
+    READ_VERSION = 4
```

Comment:
@ziafazal, this is working great!

However, I have a note about the potential performance impact of changing both `WRITE_VERSION` and `READ_VERSION`. Both [BlocksView](https://github.com/openedx/edx-platform/blob/59002b5702b84a9b1e612a65506f33de6f069fd3/lms/djangoapps/course_api/blocks/views.py#L27) and  [OutlineTabView](https://github.com/openedx/edx-platform/blob/ea6b5b9e3c5f51690858b3d8ce0106c76a0ccbf6/lms/djangoapps/course_home_api/outline/views.py#L66) are using this transformer, so it's actively read by the Learning MFE. Therefore, the block structures [will need to be updated](https://github.com/openedx/edx-platform/blob/145a30a998f76d3c5e1d391f126d31abd129bca0/openedx/core/djangoapps/content/block_structure/transformer.py#L53-L56) for all courses. According to [this doc](https://openedx.atlassian.net/wiki/spaces/AC/pages/154861855/Block+Structure+Cache+Invalidation+Proposal#BlockStructureCacheInvalidationProposal-TransformerVersionUpdates), we might need to run the management command to regenerate the cache before we start using the newer version of this transformer's data.

It would be safer if we extracted the following changes to a follow-up PR:
1. `READ_VERSION` attribute.
2. `transform`  method.
3. Tests from `lms/djangoapps/course_blocks/transformers/tests/test_hidden_content.py`.

We can observe a similar approach in #27152.

What do you think?

Also, thank you for providing a detailed explanation of this change in the PR's description (i.e. `In case of self paced course we are relying…`). Would you mind adding this paragraph to the commit's body to preserve the context?